### ISSUE TITLE
Test failing with line_profiler 4.x: for loop only reaches count of 100

### ISSUE THREAD

**Issue by juliangilbey** [spyder-ide/spyder.line_profiler#75]
With line_profiler 4.0.2, the test fails:
```
_______________________ test_profile_and_display_results _______________________

qtbot = <pytestqt.qtbot.QtBot object at 0x7fabf35dfb90>
tmpdir = local('/tmp/pytest-of-jdg/pytest-0/test_profile_and_display_resul0')

    def test_profile_and_display_results(qtbot, tmpdir):
        """Run profiler on simple script and check that results are okay."""
        os.chdir(tmpdir.strpath)
        testfilename = tmpdir.join('test_foo.py').strpath
    
        with open(testfilename, 'w') as f:
            f.write(TEST_SCRIPT)
    
        MockQMessageBox = Mock()
    
        widget = SpyderLineProfilerWidget(None)
        widget.setup()
        qtbot.addWidget(widget)
        with qtbot.waitSignal(widget.sig_finished, timeout=10000, raising=True):
            widget.analyze(testfilename)
    
        MockQMessageBox.assert_not_called()
        dt = widget.datatree
        assert dt.topLevelItemCount() == 1  # number of functions profiled
    
        top = dt.topLevelItem(0)
        assert top.data(0, Qt.DisplayRole).startswith('foo ')
        assert top.childCount() == 6
        for i in range(6):
            assert top.child(i).data(0, Qt.DisplayRole) == i + 2  # line no
    
        assert top.child(2).data(1, Qt.DisplayRole) == '1'  # hits
        assert top.child(3).data(1, Qt.DisplayRole) == '1'
>       assert top.child(4).data(1, Qt.DisplayRole) == '101'
E       AssertionError: assert '100' == '101'
E         - 101
E         + 100
```
This is because of a change of behaviour in line_profiler 4.x; using the tiny script in the package test saved to a file called `looptest.py`, I observe the following in a clean virtual environment with Python 3.9:
```
(proftest) euler:/tmp $ pip install line_profiler
Collecting line_profiler
  Downloading line_profiler-4.0.3-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (662 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 662.2/662.2 KB 6.9 MB/s eta 0:00:00
Installing collected packages: line_profiler
Successfully installed line_profiler-4.0.3
WARNING: You are using pip version 22.0.4; however, version 23.0.1 is available.
You should consider upgrading via the '/tmp/proftest/bin/python -m pip install --upgrade pip' command.
(proftest) euler:/tmp $ which kernprof
/tmp/proftest/bin/kernprof
(proftest) euler:/tmp $ kernprof -l looptest.py 
Wrote profile results to looptest.py.lprof
(proftest) euler:/tmp $ python -m line_profiler looptest.py.lprof 
Timer unit: 1e-06 s

Total time: 1.00116 s
File: looptest.py
Function: foo at line 2

Line #      Hits         Time  Per Hit   % Time  Line Contents
==============================================================
     2                                           @profile
     3                                           def foo():
     4         1    1001021.9 1001021.9    100.0      time.sleep(1)
     5         1          2.0      2.0      0.0      xs = []
     6       100         39.3      0.4      0.0      for k in range(100):
     7       100         96.8      1.0      0.0          xs = xs + ['x']

(proftest) euler:/tmp $ pip uninstall line_profiler
Found existing installation: line-profiler 4.0.3
Uninstalling line-profiler-4.0.3:
  Would remove:
    /tmp/proftest/bin/kernprof
    /tmp/proftest/lib/python3.9/site-packages/kernprof.py
    /tmp/proftest/lib/python3.9/site-packages/line_profiler-4.0.3.dist-info/*
    /tmp/proftest/lib/python3.9/site-packages/line_profiler/*
Proceed (Y/n)? y
  Successfully uninstalled line-profiler-4.0.3
(proftest) euler:/tmp $ pip install line_profiler==3.5.1
Collecting line_profiler==3.5.1
  Downloading line_profiler-3.5.1-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (70 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 70.8/70.8 KB 1.7 MB/s eta 0:00:00
Installing collected packages: line_profiler
Successfully installed line_profiler-3.5.1
WARNING: You are using pip version 22.0.4; however, version 23.0.1 is available.
You should consider upgrading via the '/tmp/proftest/bin/python -m pip install --upgrade pip' command.
(proftest) euler:/tmp $ kernprof -l looptest.py 
Wrote profile results to looptest.py.lprof
(proftest) euler:/tmp $ python -m line_profiler looptest.py.lprof 
Timer unit: 1e-06 s

Total time: 1.00129 s
File: looptest.py
Function: foo at line 2

Line #      Hits         Time  Per Hit   % Time  Line Contents
==============================================================
     2                                           @profile
     3                                           def foo():
     4         1    1001049.0 1001049.0    100.0      time.sleep(1)
     5         1          3.0      3.0      0.0      xs = []
     6       101         85.0      0.8      0.0      for k in range(100):
     7       100        150.0      1.5      0.0          xs = xs + ['x']
```

**Comment by juliangilbey** [comment]
I fixed this for Debian by changing the test as follows:
```patch
--- a/spyder_line_profiler/tests/test_lineprofiler.py
+++ b/spyder_line_profiler/tests/test_lineprofiler.py
@@ -56,7 +56,7 @@
 
     assert top.child(2).data(1, Qt.DisplayRole) == '1'  # hits
     assert top.child(3).data(1, Qt.DisplayRole) == '1' 
-    assert top.child(4).data(1, Qt.DisplayRole) == '101'
+    assert top.child(4).data(1, Qt.DisplayRole) in ['100', '101']
     assert top.child(5).data(1, Qt.DisplayRole) == '100'
 
     assert float(top.child(2).data(2, Qt.DisplayRole)) >= 900  # time (ms)
```

**Comment by jitseniesen** [comment]
Thanks for the report and the suggested fix, I will most likely do the same.

**Comment by jitseniesen** [comment]
It turns out that this fix was already applied in PR #68, so I am closing this issue.
